src/client/wieldmesh.cpp

```diff
@@ -519,51 +524,53 @@ void getItemMesh(Client *client, const ItemStack &item, ItemMesh *result)
 		result->buffer_colors.emplace_back(true, video::SColor(0xFFFFFFFF));
 		// Items with inventory images do not need shading
 		result->needs_shading = false;
+	} else if (def.type == ITEM_NODE && f.drawtype == NDT_AIRLIKE) {
+		// Fallback image for airlike node
+		mesh = getExtrudedMesh(tsrc, "no_texture_airlike.png",
+			def.inventory_overlay);
+		result->needs_shading = false;
 	} else if (def.type == ITEM_NODE) {
-		if (f.mesh_ptr[0]) {
-			mesh = cloneMesh(f.mesh_ptr[0]);
-			scaleMesh(mesh, v3f(0.12, 0.12, 0.12));
+		switch (f.drawtype) {
+		case NDT_NORMAL:
+		case NDT_ALLFACES:
+		case NDT_LIQUID:
+		case NDT_FLOWINGLIQUID: {
+			scene::IMesh *cube = g_extrusion_mesh_cache->createCube();
+			mesh = cloneMesh(cube);
+			cube->drop();
+			if (f.drawtype == NDT_FLOWINGLIQUID) {
+				scaleMesh(mesh, v3f(1.2, 0.03, 1.2));
+				translateMesh(mesh, v3f(0, -0.57, 0));
+			} else
+				scaleMesh(mesh, v3f(1.2, 1.2, 1.2));
+			// add overlays
 			postProcessNodeMesh(mesh, f, false, false, nullptr,
-				&result->buffer_colors);
-		} else {
-			switch (f.drawtype) {
-				case NDT_PLANTLIKE: {
-					mesh = getExtrudedMesh(tsrc,
-						tsrc->getTextureName(f.tiles[0].layers[0].texture_id),
-						tsrc->getTextureName(f.tiles[0].layers[1].texture_id));
-					// Add color
-					const TileLayer &l0 = f.tiles[0].layers[0];
-					result->buffer_colors.emplace_back(l0.has_color, l0.color);
-					const TileLayer &l1 = f.tiles[0].layers[1];
-					result->buffer_colors.emplace_back(l1.has_color, l1.color);
-					break;
-				}
-				case NDT_PLANTLIKE_ROOTED: {
-					mesh = getExtrudedMesh(tsrc,
-						tsrc->getTextureName(f.special_tiles[0].layers[0].texture_id), "");
-					// Add color
-					const TileLayer &l0 = f.special_tiles[0].layers[0];
-					result->buffer_colors.emplace_back(l0.has_color, l0.color);
-					break;
-				}
-				case NDT_NORMAL:
-				case NDT_ALLFACES:
-				case NDT_LIQUID:
-				case NDT_FLOWINGLIQUID: {
-					scene::IMesh *cube = g_extrusion_mesh_cache->createCube();
-					mesh = cloneMesh(cube);
-					cube->drop();
-					scaleMesh(mesh, v3f(1.2, 1.2, 1.2));
-					// add overlays
-					postProcessNodeMesh(mesh, f, false, false, nullptr,
-						&result->buffer_colors);
-					break;
-				}
-				default: {
-					mesh = createSpecialNodeMesh(client, id, &result->buffer_colors);
-					scaleMesh(mesh, v3f(0.12, 0.12, 0.12));
-				}
-			}
+				&result->buffer_colors, true);
+			break;
+		}
+		case NDT_PLANTLIKE: {
+			mesh = getExtrudedMesh(tsrc,
+			tsrc->getTextureName(f.tiles[0].layers[0].texture_id),
+			tsrc->getTextureName(f.tiles[0].layers[1].texture_id));
```

Comment:
forgive me for reviewing code style, but these need to be indented
same for plantlike_rooted below